### PR TITLE
feat(subscription): dead letter policy support

### PR DESCRIPTION
**Pull request by callmehiphop**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: system-test/pubsub.ts [hunk 1]
```diff
@@ -230,12 +230,11 @@ describe('pubsub', () => {
...
-        customAttribute: 'value',
-      };
-      const message = await publishPop(data, attrs);
-      assert.deepStrictEqual(message.data, data);
-      assert.deepStrictEqual(message.attributes, attrs);
+      const customAttribute = 'value';
+      const attrs = {customAttribute};
+      const m = await publishPop(data, attrs);
+      assert.deepStrictEqual(m.data, data);
+      assert.deepStrictEqual(m.attributes.customAttribute, customAttribute);
```
Comment (stephenplusplus) [inline system-test/pubsub.ts]: I think just `assert.strictEqual` is good here since this is now a string===string test.

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

**Review by kamalaboulhosn — APPROVED** [review]

File: system-test/pubsub.ts (hunk 1 above)
Comment (stephenplusplus) [reply system-test/pubsub.ts]: I don’t think this change was made, unless GH is caching me!

File: src/pubsub.ts [hunk 2]
```diff
@@ -338,6 +338,8 @@ export class PubSub {
    *     specified by `options.messageRetentionDuration`.
    * @property {ExpirationPolicy} [expirationPolicy] A policy that specifies
    * the conditions for this subscription's expiration.
+   * @property {DeadLetterPolicy} [deadLetterPolicy] A policy that specifies the
```
Comment (stephenplusplus) [inline src/pubsub.ts]: Could you move this up a line for alphabetical sake.

Comment (callmehiphop) [reply src/pubsub.ts]: Done!

File: system-test/pubsub.ts (hunk 1 above)
Comment (callmehiphop) [reply system-test/pubsub.ts]: Doh, done!

**Review by hongalex — APPROVED** [review]

**Review by stephenplusplus — APPROVED** [review]

**Pull request merged**